plasma/morevp_2.0.md

```diff
@@ -0,0 +1,109 @@
+All inputs are in control of a single actor `E`, so non-canonicity caused by co-transacting with a fraudster is not an issue.
+Probably it should even be blocked, so only settlements from `E` can play along in a cIFE.
+This means that there's only ever going to be one "way" to chain txs into an cIFE-ing `super_tx`.
+In the example flow given, this means in practice that `E` cannot start another cIFE starting this time from `tx2`, that would be bad!
+
+This simplifies considerations about canonicity greatly - competitors to the chained txs in a cIFE can only be found in the child chain blocks.
+
+If there's a competitor to a cIFE, it must've been provided to the child chain by a fraudulent `E`, meaning that (by RC assumption), child chain works fine.
+Now, if the child chain works fine, it means that if the chained txs in the cIFE are proven to be non-canonical, the canonical ones are sitting in the correctly propagated valid child chain blocks - users' funds are safe.
+(**TODO** if this is too weak, one can drop the assumption and strengthen the protocol by evicting the non-canonical chained txs in a cIFE).
+
+### Piggybacks of the `super_tx`
+
+Piggybacks have a couple of differences compared to MoreVP IFEs.
+
+Firstly, piggybacks are considered "jointly" for the chaining outputs like `o5` - Bob piggybacks this means that he's piggybacking an output of `tx1` and input of `tx2`.
+This is the same "money" so this makes sense.
+`o5` is then piggybacked with the same meaning, regardless of whether `tx2` is chained/disclosed by `E` or not.
+
+In particular, when Bob piggybacks `o5` before `tx2` is disclosed, Bob's bond is not slashed.
+
+### Remarks
+
+A couple of important notes:
+- **exit priority** - the age of inputs needs to be declared ahead in the step (1/), so it's actually `E.IFE(tx1, inputs_not_younger_than_x)` and priority is based on `x` (in our example flow - that of `o3`).
+During the process of chaining, there can be no inputs "prepended" to the `super_tx` (what happened to `o3`) that are younger than `x`.
+In other words, `E` provides an _a priori_ upper bound on exit priority.
+This is because we cannot easily go about updating the priority queue. If not for `x` declared, we'd been stuck with priority of `o2`, which is the youngest input at time when `E.IFE(tx1...` is done
+- **transaction balance** - i.e. the `sum(inputs)>=sum(outputs)` is really tracked for the whole tx - if all chaining txs satisfy it, so does the entire `super_tx`
+- **unmatched funds on the exchange** - in a variant of the flow, when the chain of txs ends with `(E, D)` holding funds, `D` piggybacks that and `D` will get the payout (see piggybacking)
+- **gas** - cIFE should incur no more gas costs than IFEing all the chained txs, so if we're fine with gas costs of IFE for payment, we should be too with costs of cIFE for settlement (**TODO** - verify this, this is intuition)
```

Comment:
I think the different is the possible size of an atomic swap. Chaining (in here) != atomic, but I think in practice it is fine.

It does lead to some problem as the batch size per tx is more limited in MoreVP, how do we break a settlement into multiple txs? Previously we can always have a big tx to do huge atomic swap on settlement. Now we need to break that huge one into several small ones. For continuous trading it seems more easy and less of an issue as per settlement size is expected to be smaller. However, in batch auction how do we break a huge batch into small chained txs would need some design.

Reply:
yep, chaining isn't atomic. Also continuous trading was the objective here. Batch settlement that must be atomic probably doesn't fit here.

But, did it fit under MVP? If the batch settlement was split across many MVP txs, they still can be non-atomic, if they end up spanning >1 child block.

Reply:
For batch auction I guess the most practical/simple solution we would do is just a large tx in MVP that settles everything. However, if we are to break it into chain computations, and we want to do cross-block atomic settlement, I believe we can design the chaining to be cross blocks by explicit a `chain-id` that need to be singed as confirm sig. All chained txs need to have such `chain-id` data explicitly.

anyway, now I understand that your target is of continuous trading. However, if we are taking this and abandon MVP in ALD, would this means we are closing our door to batch auction too?